extensions/proposals/WEBGL_element_texture/extension.xml

```diff
@@ -0,0 +1,298 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<proposal href="proposals/WEBGL_element_texture/">
+  <name>WEBGL_element_texture</name>
+
+  <contact> <a href="https://www.khronos.org/webgl/public-mailing-list/">WebGL
+  working group</a> (public_webgl 'at' khronos.org) </contact>
+
+  <contributors>
+    <contributor>Byungseon Shin, LG Electronics</contributor>
+
+    <contributor>Andrei Volykhin, LG Electronics</contributor>
+
+    <contributor>Mark Callow, HI Corporation</contributor>
+
+    <contributor>Members of the WebGL working group</contributor>
+  </contributors>
+
+  <number>NN</number>
+
+  <depends>
+    <api version="1.0"/>
+  </depends>
+
+  <overview>
+      <p>This extension defines a new object, the <code>HTMLElementTexture</code>,
+    that can be used to efficiently transfer a sequence of image frames from
+    a producer which out of control the WebGL into a WebGL texture.
+    This is done via a new texture target, <code>TEXTURE_ELEMENT_WEBGL</code>
+    which can only be specified as being the consumer of an image stream from a
+    producer element.</p>
+
+    <p>There is no support for most of the functions that manipulate other
+    texture targets (e.g. you cannot use <code>*[Tt]ex*Image*()</code>
+    functions with <code>TEXTURE_ELEMENT_WEBGL</code>). Also,
+    <code>TEXTURE_ELEMENT_WEBGL</code> targets never have more than a single
+    level of detail. Because of these restrictions, it is possible to
+    allow sources which have internal formats not otherwise supported by WebGL,
+    such as planar or interleaved YUV data to be WebGL texture target siblings.</p>
+
+    <p>The extension extends GLSL ES with a new <code>samplerElementWebGL</code> type
+    and matching sampling functions that provide a place for an implementation
+    to inject code for sampling non-RGB data when necessary without degrading performance
+    for other texture targets. Sampling a <code>TEXTURE_ELEMENT_WEBGL</code> via a sampler of
+    type <code>samplerElementWebGL</code> always returns RGBA data.</p>
+
+    <p>Sampling a WebGL texture will return an RGBA vector in the same
+    colorspace as the source image. If the source image is stored in YUV
+    (or some other basis) then the YUV values will be transformed to RGB values.
+    (But these RGB values will be in the same colorspace as the original image.
+    Colorspace here includes the linear or non-linear encoding of the samples.
+    For example, if the original image is in the sRGB color space then the RGB value
+    returned by the sampler will also be sRGB, and if the original image is stored
+    in ITU-R Rec. 601 YV12 then the RGB value returned by the sampler will be
+    an RGB value in the ITU-R Rec. 601 colorspace)</p>
```

Comment:
As I wrote in #2508, this decision was made before sRGB support in OpenGL ES. If we continue with this approach a query for either the colorspace name or the actual transfer function will be needed so apps can convert to linear data for blending and can convert properly to the canvas color space.

We should consider having this extension return samples in linear space instead. It should only be a few more instructions in the injected shader code.

Reply:
Ok, i will remove this part from proposal and instead add issue for linear vs non-linear color space.